### DISCUSSION TITLE
Unable to start codespace. Please try again later.

### DISCUSSION THREAD

**Discussion by ram-if** [community/community#123358] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

I have been using a codespace with a few uncommitted changes. When I try to open it today, its repeatedly failing to open with the error - "Unable to start codespace. Please try again later." And the codespaces homepage is showing the codespace as active, which means my usage hours are being consumed.

Did anyone else have the same problem? Please suggest what to do

**Comment by ram-if** [comment]
I tried "export changes to a branch" option, but it failed with the error - "Your codespace export did not complete. Please try again."  Whats worse is that my uncommitted changes seems to have lost.. Earlier the codespace use to show the message - "This codespace has uncommitted changes", now it shows "No changes"

**Comment by danala04** [comment]
I'm facing the same issue, I also need to retrieve my uncommitted changes.
![image](https://github.com/community/community/assets/87354757/0f090e0e-1669-491e-979a-5d49a5284228)

**Comment by keyanna1501** [comment]
i am also having this issue...

**Comment by MS-Jahan** [comment]
![image](https://github.com/community/community/assets/27774290/baef8721-0527-4fbf-ae48-2e117077586a)
Same. Though I was able to "Export changes to a branch".

**Comment by harshh-techqware** [comment]
Same for me "Unable to start codespace. Please try again later."

**Comment by natitack** [comment]
Adding my name to the list. I should add, that I get this error trying to open any codespace at all, not just if I am trying to export changes.

**Comment by okwesil** [comment]
Same for me as well

**Comment by kanata-okayama** [comment]
me too

**Comment by Patil191** [comment]
same here

**Comment by deadbeef-development** [comment]
go üp

**Comment by hotateo141** [comment]
me too

**Comment by chrischeuk** [comment]
same here

**Comment by developertimm** [comment]
Same here my project i have to push to prod

**Comment by developertimm** [comment]
New code space are starting but old code spaces are not showing error

**Comment by 1DavidRM** [comment]
Me too, I wanna cry, I need to finish my proyect for this nighttttt, pls someone helppp

**Reply by ITILD** [reply]
https://docs.github.com/en/codespaces/developing-in-a-codespace/using-github-codespaces-in-visual-studio-code

can not connect web
with vscode desktop it work

**Comment by egamit** [comment]
me too

**Comment by apsion-test** [comment]
Same

**Comment by krishealty** [comment]
There can be multiple reasons for it, but if so many people are facing the same issue, it can be an internal server error, still i have listed some of the issues that can be resolved:

So usually, codespaces are set to stop after 30 minutes without any activity. If you try to interact with a codespace after it has stopped, you may see a `503 service unavailable error`.

If a Start button is shown in Visual Studio Code or in your browser window, click Start to reconnect to the codespace.
Reset your codespace by reloading the window. From the Command Palette in Visual Studio Code, click Developer: Reload Window.

Sometimes you may not be able to access a codespace from your browser. If this happens, go to [codespace link](https://github.com/codespaces) and try connecting to the codespace from that page.

If the codespace is not listed on that page, check that you are the owner of the codespace you are trying to connect to. You can only open a codespace that you created.
If the codespace is listed but you cannot connect from that page, check whether you can connect using a different browser.

If the codespace seems particularly slow or has latency issues, it is possible that it has been created in a region that is far from you. To resolve this, you can manually set your [GitHub Codespaces region.](https://docs.github.com/en/codespaces/setting-your-user-preferences/setting-your-default-region-for-github-codespaces)

The same question was asked on Stack Overflow, here's the [answer](https://stackoverflow.com/a/77050124) and [full discussion.](https://stackoverflow.com/questions/76998936/no-longer-able-to-connect-to-codespace)

You can also refer to the [Official documentation](https://docs.github.com/en/codespaces/troubleshooting/working-with-support-for-github-codespaces) and troubleshoot your problem, thanks.

**Reply by deadbeef-development** [reply]
@krishealty
Are codespaces working for you?

**Comment by kristynlynn** [comment]
Also experiencing this issue. If I create new codespaces, I can stop and restart them with any issue. I tried 4 of my codespaces that were created May 9 or earlier and none of them work. Tried multiple browsers (Edge and Firefox) as well but browser didn't seem to make a difference.

**Comment by ram-if** [comment]
The github status page shows codespaces in green, and there is no incident tracked yet :(

**Comment by kbegiedza** [comment]
Still occurs...

**Comment by AnthonyFrancis** [comment]
Same for me

**Comment by LordRyanLee** [comment]
Same here....

**Comment by Ekesmall** [comment]
Same here too oo... The problem is general

**Comment by ricecin** [comment]
Same here, the issue is also showing up on Harvard's CS50 status page.

**Comment by logouter024516** [comment]
In S,korea, Same here.....
And I can use codespace at 10:30am

**Reply by howardwu1** [reply]
in south korea for vacation as well and also seeing the issue :)

**Comment by chemlord** [comment]
Having this issue in Toronto, Canada. Glad I'm not the only one

**Comment by HannahNely** [comment]
I got so frustrated because I thought it was something I did wrong. Im having the same issue!

**Comment by buildding** [comment]
me too

**Comment by scottdonnelly** [comment]
Joining the club.  None will open.  Not cool.

**Comment by bandana-edtech** [comment]
I am also having the same issue. But github status page is showing no issues with Codespace

**Comment by qqwqy** [comment]
Me too. So it seems that google colab is more stable

**Comment by WELLMAX357** [comment]
Same error in Japan.

**Comment by PoULkY** [comment]
Same here. Yesterday I can use it with no problem.

**Reply by LDN85** [reply]
So is it to me!

**Comment by nikhilx144** [comment]
Same here. Could someone please help me?

**Comment by ChrisBeil** [comment]
same error in germany

**Comment by CS50github2024** [comment]
Same in Australia

**Reply by CS50github2024** [reply]
I'd be happy to work offline, however harvard's cs50.h library in C is not on my machine and I have no idea how to include it with clang so I can include it when I compile.  It's only codespace not working all else github works as far as I can tell.

**Comment by ahmedbentabet** [comment]
Hey, Morocco is here :)

**Comment by marchingon12** [comment]
i thought i was the only one 😭 thank goodness. This happened to me yesterday and the day before as well.

**Comment by Yuan0100** [comment]
I found it works to use codespace with VS Code Desktop (with the codespace extension installed).

**Reply by ram-if** [reply]
Yup, this is working! Thanks for the tip!

**Comment by anshulpatel25** [comment]
I am also facing the same issue. Gujarat, India

**Comment by Xuhhhh52071** [comment]
same in China

**Reply by ITILD** [reply, marked as answer]
https://docs.github.com/en/codespaces/developing-in-a-codespace/using-github-codespaces-in-visual-studio-code

this can work

**Reply by hvl71** [reply]
Same issue here since last night. I can create a new code space via https://github.com/codespaces/new but not open existing ones. Only relevant if no uncommitted changes or they are not important..

**Reply by plemnoooooo** [reply]
> https://docs.github.com/en/codespaces/developing-in-a-codespace/using-github-codespaces-in-visual-studio-code
> 
> 
> 
> this can work

@ram-if I believe this answer is not correct; this article only shows how to create a Github Codespace and how to configure it. It does not show a method to fix the issue. Un-mark the answer so that Github knows that the issue still persists.

**Reply by ram-if** [reply]
@plemnoooooo Agree, it doesnt fix the issue per se.. but its a workaround. Please follow the instructions in "Prerequisites" and "Opening a codespace in VS Code" sections of the article. You should be able to access your codespace from VS code.. It worked for me

**Comment by victorkingi** [comment]
Same here, Manchester UK

**Reply by aavashsubedi** [reply]
Same here Manchester UK

**Comment by jstoneman-ME** [comment]
Me too - Melbourne, Australia. What's up - anyone heard anything?

**Reply by mingmamma** [reply]
Hey fellow Melbournian...was banging my head against the wall til seeing this post~~

**Comment by gabrielesqvc** [comment]
Same here, I'm trying to finish my last assignments .

**Comment by mingmamma** [comment]
Reposting here to a suggestion given above to get noticed, working for my end just a minute ago:
https://github.com/orgs/community/discussions/123358#discussioncomment-9393514

**Reply by dsetzer** [reply]
that's not a solution..

**Reply by mikehudson2** [reply]
Same here, London

**Reply by junkmd** [reply]
Same here, Tokyo

**Reply by danala04** [reply]
same i still cant retrieve my uncommited changes, it still error

**Reply by ram-if** [reply]
It looks like the browser interface of codespaces is not working. I am able to open my codespaces (with uncommitted changes) via  "Github codespaces" extension on my local VS code.. Ofcourse, this is a temporary workaround

**Reply by RFebrians** [reply]
same here , Jakarta

**Reply by mikehudson2** [reply]
Solar storm maybe!

**Reply by mingmamma** [reply]
Umm ... individual experiences may vary but just posting an image for extra clarity of what seemed to work for me: accessing Codespaces from local VsCode, enabled by the Codespaces VsCode extension
![Screen Shot 2024-05-11 at 19 15 27](https://github.com/community/community/assets/18562714/4f157621-6209-4f7b-97c1-79353c4b2123)
I think that's the appoach the link refers to though I didn't read carefully

**Comment by wgilpin** [comment]
Me too. Shows the risk of web hosted dev environments I guess

**Comment by angegar** [comment]
Same in France. I noticed it yesterday evening.

![image](https://github.com/community/community/assets/5628764/c38ec60a-cc31-4e81-8151-be5837fdcc9b)

**Comment by hssumar** [comment]
Same here.

**Comment by aavashsubedi** [comment]
![image](https://github.com/community/community/assets/87420856/251ffdef-3b94-46fc-ac8f-3a6e878b1579)

**Comment by carlesViolan** [comment]
> ### Select Topic Area
> Bug
> 
> ### Body
> I have been using a codespace with a few uncommitted changes. When I try to open it today, its repeatedly failing to open with the error - "Unable to start codespace. Please try again later." And the codespaces homepage is showing the codespace as active, which means my usage hours are being consumed.
> 
> Did anyone else have the same problem? Please suggest what to do

same in Barcelona!

**Comment by pdparchitect** [comment]
8+ hour outage yet status page is all green. Somebody at GitHub dropped the ball!

**Comment by seonwee** [comment]
same in china

**Comment by joshaber** [comment]
Hello from GitHub 👋

Sorry folks, this was a miss on our part. It should be resolved now and we’ll be doing an internal retrospective on how we can improve going forward.

**Reply by ram-if** [reply]
Its working for me now. Thanks

**Reply by nikhilx144** [reply]
Its working now. Thanks!

**Reply by junkmd** [reply]
I have confirmed that it works in the browser. Thank you.

**Reply by kristynlynn** [reply]
Thank you. It does in fact seem to be working now. Should I expect to see something added to the Github status page's incident history then? I still don't see anything listed there.

**Comment by nikhilx144** [comment]
Its working now

**Comment by Lewis-Gu** [comment]
Codespace is still not working even today..

**Reply by plemnoooooo** [reply]
It might be a firewall issue on your end. You may want to read on this article first: https://docs.github.com/en/codespaces/troubleshooting/troubleshooting-your-connection-to-github-codespaces

**Comment by demovoidgan** [comment]
me too today
